SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+                $NewAppVolume = $NewAppVolume.Substring(0, $NewAppVolume.Length - 1)
+                $AppVolume = $NewAppVolume
+                Write-Host "AppVolume moved to " $AppVolume
+            }
+            else {
+                $AppVolume = $NewAppVolume
+                Write-Host "AppVolume moved to " $AppVolume
+            }
+            If ([string]::IsNullOrEmpty($NewAppVolume)) {
+                Write-Host "AppVolume remains on " $AppVolume
+            }
+
+            Write-Host "BackupVolume: " $BackupVolume
+            $NewBackupVolume = Read-Host "Your BackupVolume: "
+            If ($NewBackupVolume.Substring($NewBackupVolume.Length - 2 -eq $CheckLastTwoChar) -and $NewBackupVolume.Length -gt 2) {
+                $NewBackupVolume = $NewBackupVolume.Substring(0, $NewBackupVolume.Length - 2)
+                $BackupVolume = $NewBackupVolume
+                Write-Host "BackupVolume moved to " $BackupVolume
+            }
+            elseif ($NewBackupVolume.Substring($NewBackupVolume.Length - 1 -eq $CheckLastChar) -and $NewBackupVolume.Length -gt -1) {
+                $NewBackupVolume = $NewBackupVolume.Substring(0, $NewBackupVolume.Length - 1)
+                $BackupVolume = $NewBackupVolume
+                Write-Host "BackupVolume moved to " $BackupVolume
+            }
+            else {
+                $BackupVolume = $NewBackupVolume
+                Write-Host "BackupVolume moved to " $BackupVolume
+            }
+            If ([string]::IsNullOrEmpty($NewBackupVolume)) {
+                Write-Host "BackupVolume remains on " $BackupVolume
+            }
+        }
+        Default {Write-Host "Drives agreed, continuing"; }
+    }
+
+    $CheckLastTwoChar = ":\"
+    $CheckLastChar = "\"
+
+    $SetupFile = Read-Host -Prompt 'Please enter the root location for Setup.exe'
+    IF ($SetupFile.Length -gt 1) {
+        $C2 = $SetupFile.Substring($SetupFile.Length - 2)
+        $C1 = $SetupFile.Substring($SetupFile.Length - 1)
+        If ($C2 -eq $CheckLastTwoChar) {
+            $debug = $SetupFile.Substring($SetupFile.Length - 2)
+            Write-Host $debug '/' $CheckLastTwoChar
+            $SetupFile = $SetupFile.Substring(0, $SetupFile.Length - 2)
+            Write-Host $SetupFile
+        }
+        elseif ($C1 -eq $CheckLastChar) {
+            $SetupFile = $SetupFile.Substring(0, $SetupFile.Length - 1)
+            Write-Host $SetupFile
+        }
+    }
+    IF ($SetupFile.Length -eq 1) {
+        $SetupFile = $SetupFile + ':\SQLEXPR_x64_ENU\SETUP.EXE'
+        Write-Host 'Setup will start from ' + $SetupFile
+    } 
+    else {
+        $SetupFile = $SetupFile + '\SQLEXPR_x64_ENU\SETUP.EXE'
+        Write-Host 'Setup will start from ' + $SetupFile
+    }
+
+
+    $ConfigFile = 'c:\temp\'
+
+    if ( -Not (Test-Path -Path $ConfigFile ) ) {
+        New-Item -ItemType directory -Path $ConfigFile
+    }
+
+    Out-File -FilePath C:\Temp\ConfigurationFile2.ini -InputObject $startScript
```

Comment:
C:\temp may not exist. Since `$env:temp` is formatted in DOS syntax, use `[System.IO.Path]::GetTempPath()`